tests/src/Context/DrupalContext.php

```diff
@@ -156,4 +152,92 @@ public function assertUserExistence($username) {
     }
   }
 
+  /**
+   * Creates memberships for the provided entities.
+   *
+   * If the member type is user, the username should be passed instead.
+   *
+   * @param TableNode $membership_table
+   *   The membership table.
+   *
+   *   Table format:
+   *   | group_type   | group_id  | member    |
+   *   | entityTypeId | entity_id | user name |
+   *   | entityTypeId | entity_id | user name |
+   *
+   *   Only the name field is required.
+   *
+   * @throws \Exception
+   *   Thrown when an entity is not found or when an entity set as group is not
+   *   a group.
+   *
+   * @Given (the following )user memberships:
+   */
+  public function givenUserMembership(TableNode $membership_table) {
+    foreach ($membership_table->getColumnsHash() as $membership_row) {
+      foreach ($membership_row as $key => $value) {
+        $values[$key] = $value;
+      }
+
+      // Load group.
+      $group = \Drupal::entityTypeManager()
+        ->getStorage($values['group_type'])
+        ->load($values['group_id']);
+      if (empty($group)) {
+        throw new \Exception("Entity " . $values['group_id'] . " not found.");
+      }
```

Comment:
This loads every group one by one in a loop. It would be more efficient to load all at once using `loadMultiple()`.